src/shared/extension.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export {}
+
+// Causing colorization issues if placed above Array.prototype...
+// Ideally: ((_) => number) | ((_) => string)
+type Selector<T> = (_: T) => number | string
+
+declare global {
+	interface ArrayConstructor {
+		commonLength(a: any[], b: any[]): number
+	}
+	interface Array<T> {
+		last: T
+		replace(items: T[]): void // From Mobx, but not showing up.
+		remove(item: T): boolean // From Mobx, but not showing up.
+		removeWhere(predicate: (item: T) => boolean): T | false
+		sortBy<T>(this: T[], selector: Selector<T>, descending?: boolean): Array<T> // Not a copy
+	}
+	interface String {
+		file: string
+		path: string
+	}
+}
+
+!Array.hasOwnProperty('commonLength') &&
+Object.defineProperty(Array, 'commonLength', {
+	value: function(a: any[], b: any[]): number {
+		let i = 0
+		for (; a[i] === b[i] && i < a.length && i < b.length; i++) {}
+		return i
+	}
+})
+
+!Array.prototype.hasOwnProperty('last') &&
+Object.defineProperty(Array.prototype, 'last', {
+	get: function() {
+		return this[this.length - 1]
+	}
+})
+
+!Array.prototype.hasOwnProperty('removeWhere') &&
+Object.defineProperty(Array.prototype, 'removeWhere', {
+	value: function(predicate: (item: any) => boolean) { // Unable to express (item: T) so using (item: any).
+		const i = this.findIndex(predicate)
+		return i >= 0 && this.splice(i, 1).pop()
+	}
+})
+
+Array.prototype.sortBy = function<T>(selector: Selector<T>, descending = false) {
+	this.sort((a, b) => {
+		const aa = selector(a)
+		const bb = selector(b)
+		const invert = descending ? -1 : 1
+		if (typeof aa === 'string' && typeof bb === 'string') return invert * aa.localeCompare(bb)
+		if (typeof aa === 'number' && typeof bb === 'number') return invert * (aa - bb)
+		return 0
+	})
+	return this
+}
+
+!String.prototype.hasOwnProperty('file') &&
+Object.defineProperty(String.prototype, 'file', {
+	get: function() {
+		return this.substring(this.lastIndexOf('/') + 1, this.length)
```

Comment:
>/ [](start = 42, length = 1)

does this work on Windows?

Reply:
Need to fix. Thanks for the catch. There is some additional windows testing we (likely just Mukul and I) need to do. It was difficult earlier as I am on a Mac.

Reply:
I think this is just path.basename

---
In reply to: [434777721](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r434777721) [](ancestors = 434777721)

Reply:
See comments for "path.dirname".

---
In reply to: [435599059](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r435599059) [](ancestors = 435599059,434777721)

Reply:
This will be addressed.

---
In reply to: [434773253](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r434773253) [](ancestors = 434773253)

Reply:
Unfortunately not.

---
In reply to: [438342421](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r438342421) [](ancestors = 438342421,436180678,435599059,434777721)